src/main/java/com/southernartsoap/controller/MainController.java

```diff
@@ -8,49 +9,59 @@
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 
 import com.southernartsoap.model.Product;
 import com.southernartsoap.service.ProductService;
 
 @Controller
-@ControllerAdvice // This makes the `@ModelAttribute`s of this controller available to all controllers, for the navbar.
+@ControllerAdvice // This makes the `@ModelAttribute`s of this controller available to all
+					// controllers, for the navbar.
 public class MainController {
-    @Autowired
-    ProductService productService;
-
-    @GetMapping(value = "/")
-    public String home() {
-        return "index";
-    }
-    
-    @ModelAttribute(value = "products")
-    public List<Product> products() {
-        return productService.findAll();
-    }
-
-    @ModelAttribute(value = "categories")
-    public List<String> categories() {
-        return productService.findDistinctCategories();
-    }
-
-    @GetMapping(value = "/products/filter")
-    public String filter(@RequestParam(required = false) String category,
-                         Model model) {
-        List<Product> filtered = productService.findByCategory(category);
-        model.addAttribute("products", filtered); // Overrides the @ModelAttribute above.
-        return "filters";
-    }
-    
-    @GetMapping(value = "/products")
-    public String shopAll() {
-        return "filters";
-    }
-
-    @GetMapping(value = "/about")
-    public String about() {
-        return "about";
-    }
-}
+	@Autowired
+	ProductService productService;
+
+	@GetMapping(value = "/")
+	public String home() {
+		return "index";
+	}
+
+	@ModelAttribute(value = "products")
+	public List<Product> products() {
+		return productService.findAll();
+	}
+
+	@ModelAttribute(value = "categories")
+	public List<String> categories() {
+		return productService.findDistinctCategories();
+	}
```

Comment:
why are these in the controller? I thought controller was exclusively for mappings?